### ISSUE TITLE
Date Picker Constraint Not Enforced on 'Ok' Button Click.

### ISSUE THREAD

**Issue by AdityaKBhadragond14** [google/android-fhir#2508]
**Describe the bug**
In the date picker if there is some constraint which does not allow to select the current date and the user still clicks on "OK" button without selecting any date it still selects the current date as the answer as it is the default to it.

[Screen_recording_20240404_153456.webm](https://github.com/google/android-fhir/assets/102521117/dc99a15b-a44e-4acb-88c0-dbe59565ca0e)

**Component**
SDC Gallery app

**To Reproduce**

**Expected behavior**
There can be two approaches to solving this:
1. Disable the "OK" button if the date selected is invalid enable only when the selected date is valid.
2. Set the default date to the minimum valid date. This way we can reduce the number of clicks the user has to do in order to select a valid date.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: [e.g. Pixel4a emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]

**Additional context**

**Would you like to work on the issue?**
Yes.

**Comment by AdityaKBhadragond14** [comment]
@MJ1998 Please check if I have missed anything to add here. Thanks.

**Comment by MJ1998** [comment]
I would say disable the calendar popup itself when there is no valid date possible + show an error that there is no valid date.
Thanks Aditya.

**Comment by AdityaKBhadragond14** [comment]
> I would say disable the calendar popup itself when there is no valid date possible + show an error that there is no valid date. Thanks Aditya.

Okay, this will be for the case when the min and max dates are overlapping, right? Also I think going with 2nd solution to solve this issue would be better as you had suggested because this will avoid increasing the number of clicks.

**Comment by MJ1998** [comment]
Yes right. I suggested this `I would say disable the calendar popup itself when there is no valid date possible + show an error that there is no valid date.` only when max < min.
The other solution of yours seems correct!

**Comment by AdityaKBhadragond14** [comment]
Okay got it.

**Comment by jingtang10** [comment]
> I would say disable the calendar popup itself when there is no valid date possible + show an error that there is no valid date. Thanks Aditya.

personally not too keen on this - it's such a niche case and would probably mean the questionnaire author needs to do something about it - if the behaviour of our widget is consistent (i guess you would not be able to input any date ultimately) i'm happy for us to not handle this case additionally.
